web/styles/inputs.css

```diff
@@ -48,16 +48,18 @@
     }
 
     &.has-input-icon .input-element {
+        /* Subtract 1px to compensate for the left border */
         padding-left: calc(
             var(--input-icon-starting-offset) + var(--input-icon-width) +
-                var(--input-icon-content-gap)
+                var(--input-icon-content-gap) - 1px
```

Comment:
This complexity makes me wonder if we could use an `outline` rather than a border.